rllib/agents/alpha_star/alpha_star.py

```diff
@@ -292,9 +293,10 @@ def training_iteration(self) -> ResultDict:
                 remote_fn=self._sample_and_send_to_buffer,
             )
         # Update sample counters.
-        for (env_steps, agent_steps) in sample_results.values():
-            self._counters[NUM_ENV_STEPS_SAMPLED] += env_steps
-            self._counters[NUM_AGENT_STEPS_SAMPLED] += agent_steps
+        for sample_result in sample_results.values():
+            for (env_steps, agent_steps) in sample_result:
+                self._counters[NUM_ENV_STEPS_SAMPLED] += env_steps
```

Comment:
ah so what we're doing here, is updating the `NUM_ENV_STEPS_SAMPLED` counter every time samples are returned by the sampling workers. I could also do this on the outside of this loop, but chose to do it on the inside since I needed the `self._counters[NUM_ENV_STEPS_SAMPLED]` to update the global state of the sampling workers. The global state is used for stuff like anealing learning rates, etc, so it doesn't necessarily to our vanilla apex implementation, but could matter for additional mixins and callbacks

Reply:
Yeah, seems more accurate to do this inside. Cool.